eZ/Publish/API/Repository/Tests/SearchServiceTest.php

```diff
@@ -656,6 +657,134 @@ public function getContentQuerySearches()
+                    'limit' => 50,
+                ],
+                $fixtureDir . 'UserIdEq.php',
+            ],
+            [
+                [
+                    'query' => new Criterion\UserId([14, 10]),
+                    'sortClauses' => [new SortClause\ContentId()],
+                    'limit' => 50,
+                ],
+                $fixtureDir . 'UserIdIn.php',
+            ],
+            [
+                [
+                    'query' => new Criterion\UserLogin('*adm*', Operator::LIKE),
+                    'sortClauses' => [new SortClause\ContentId()],
+                    'limit' => 50,
+                ],
+                $fixtureDir . 'UserLogin.php',
+            ],
+            [
+                [
+                    'query' => new Criterion\UserLogin('admin', Operator::EQ),
+                    'sortClauses' => [new SortClause\ContentId()],
+                    'limit' => 50,
+                ],
+                $fixtureDir . 'UserLogin.php',
+            ],
+            [
+                [
+                    'query' => new Criterion\UserLogin(['admin'], Operator::IN),
+                    'sortClauses' => [new SortClause\ContentId()],
+                    'limit' => 50,
+                ],
+                $fixtureDir . 'UserLogin.php',
+            ],
+            [
+                [
+                    'query' => new Criterion\UserEmail('*nospam*', Operator::LIKE),
+                    'sortClauses' => [new SortClause\ContentId()],
+                    'limit' => 50,
+                ],
+                $fixtureDir . 'UserEmail.php',
+            ],
+            [
+                [
+                    'query' => new Criterion\UserEmail('nospam@ez.no', Operator::EQ),
+                    'sortClauses' => [new SortClause\ContentId()],
+                    'limit' => 50,
+                ],
+                $fixtureDir . 'UserEmail.php',
+            ],
+            [
+                [
+                    'query' => new Criterion\UserEmail(['nospam@ez.no'], Operator::IN),
+                    'sortClauses' => [new SortClause\ContentId()],
+                    'limit' => 50,
+                ],
+                $fixtureDir . 'UserEmail.php',
+            ],
+            [
+                [
+                    'query' => new Criterion\SectionIdentifier('users'),
+                    'sortClauses' => [new SortClause\ContentId()],
+                    'limit' => 50,
+                ],
+                $fixtureDir . 'SectionIdentifier.php',
+            ],
+            [
+                [
+                    'query' => new Criterion\SectionIdentifier(['users']),
+                    'sortClauses' => [new SortClause\ContentId()],
+                    'limit' => 50,
+                ],
+                $fixtureDir . 'SectionIdentifier.php',
+            ],
+            [
+                [
+                    'query' => new Criterion\ObjectStateIdentifier('not_locked'),
```

Comment:
I think we already discussed internally also this issue. If you have an object state which shares the same identifier across multiple groups, then this is not very useful. The case might not be as uncommon as you might think.

Reply:
This has been fixed shortly after the demo when you DMed me. It is that the tests are not relying on it. 

ref.  https://github.com/ezsystems/ezpublish-kernel/pull/2952/files#diff-b300f6e65c848771c4916553315ecc83R65

Reply:
fwiw, in Mig Bundle we do allow usage of state identifier, as it is pretty nice to be able to skip the group identifier. Then the code checks: if the identifier is unique, all god. If it is reused across groups, it throws.